src/radiation/mocmc.cpp

```diff
@@ -0,0 +1,1100 @@
+// © 2021-2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+
+#include <globals.hpp>
+#include <kokkos_abstraction.hpp>
+#include <utils/error_checking.hpp>
+
+// singularity
+#include <singularity-eos/eos/eos.hpp>
+#include <singularity-opac/neutrinos/opac_neutrinos.hpp>
+
+#include "closure.hpp"
+#include "closure_mocmc.hpp"
+#include "geodesics.hpp"
+#include "kd_grid.hpp"
+#include "phoebus_utils/interpolation.hpp"
+#include "phoebus_utils/root_find.hpp"
+#include "radiation/radiation.hpp"
+#include "reconstruction.hpp"
+
+namespace radiation {
+
+namespace pf = fluid_prim;
+namespace cf = fluid_cons;
+namespace cr = radmoment_cons;
+namespace pr = radmoment_prim;
+namespace ir = radmoment_internal;
+namespace im = mocmc_internal;
+using namespace singularity;
+using namespace singularity::neutrinos;
+using singularity::RadiationType;
+using vpack_types::FlatIdx;
+
+constexpr int MAX_SPECIES = 3;
+
+// TODO(BRR) add options
```

Comment:
What's the plan for the optionality? Not that we need to think too hard about that for now.

Reply:
One reason I ask is I'm wondering how we will pass those settings through into the kernel.

Reply:
I'm not too sure what options we want beyond this are. But eventually I think we'll want multiple functions, including ones more sophisticated than just the uniform-over-all-zones function here.

Reply:
kk nbd for now